src/main/java/io/nats/client/impl/NatsConnectionWriter.java

```diff
@@ -53,9 +53,16 @@
         int bufSize = connection.getOptions().getBufferSize();
         this.sendBuffer = new byte[bufSize];
 
-        // Limit the messages that we can push on the queue to 5x the number we would send in a single batch
+        // The value of 1000 is arbitrary, based on performance testing, a send buffer is also configured above and
+        // that size is used first, only if many small messages are being sent on a big buffer will this value
+        // come into play.
         this.maxAccumulate = 1000;

```

Comment:
Still think tracking pending outbound bytes is best long term and benchmark program can watch that.

Reply:
we can't use bytes with the linked blocking queue, since we aren't the ones waiting. The old code could make that distinction but the linked queue just blocks on the size in "slots".

We could muck with it, but I then we are back to the custom implementation.